pkg/lib/cockpit.d.ts

```diff
@@ -368,4 +368,134 @@ declare module 'cockpit' {
 
     export let localStorage: Storage;
     export let sessionStorage: Storage;
+
+    /**
+     * Options passed to the HTTP client constructor.
+     */
+    export interface HttpOptions {
+        /**
+         * The target address. If omitted, the endpoint string must include the host.
+         */
+        address?: string;
+        /**
+         * The port to connect to.
+         */
+        port?: number;
+        /**
+         * Enable TLS (HTTPS) connection.
+         */
+        tls?: boolean;
```

Comment:
This isn't a boolean, but an object which gives the cert/key, validation option etc. Please see the description and example in https://cockpit-project.org/guide/latest/cockpit-http.html